File: chapters/01-basic-syntax/examples/programmer.go
// Go版本的程序员示例
package main

import "fmt"

type Programmer struct {
	Name     string
	Language string
}

func NewProgrammer(name, language string) *Programmer {
	return &Programmer{
		Name:     name,
		Language: language,
	}
}

func (p *Programmer) Introduce() string {
	return fmt.Sprintf("Hi, I'm %s and I love %s!", p.Name, p.Language)
}

func main() {
	gopher := NewProgrammer("Alice", "Go")
	fmt.Println(gopher.Introduce())
}

File: chapters/01-basic-syntax/examples/programmer.rs
// Rust版本的程序员示例
struct Programmer {
    name: String,
    language: String,
}

impl Programmer {
    fn new(name: String, language: String) -> Programmer {
        Programmer { name, language }
    }
    
    fn introduce(&self) -> String {
        format!("Hi, I'm {} and I love {}!", self.name, self.language)
    }
}

fn main() {
    let rustacean = Programmer::new(
        String::from("Alice"), 
        String::from("Rust")
    );
    println!("{}", rustacean.introduce());
}